hera_sim/interpolators.py

```diff
@@ -131,3 +157,106 @@ def _check_pol(self, pol):
+            or 'interp1d'. If 'poly1d' is chosen, then ref_file must be a 
+            .npy file. If 'interp1d' is chosen, then ref_file must be a .npz 
+            file with two arrays stored in its archive: 'freqs' and 'beam'.
+
+        **interp_kwargs (dict, optional):
+            Keyword arguments to be passed to the interpolator in the case that
+            'interp1d' is chosen.
+
+    Raises:
+        AssertionError:
+            This is raised if the choice of interpolator and the required type
+            of the ref_file do not agree (i.e. trying to make a 'poly1d' object
+            using a .npz file as a reference). An AssertionError is also raised
+            if the .npz for generating an 'interp1d' object does not have the
+            correct arrays in its archive.
+    """
+
+    def __call__(self, freqs):
+        return self._interpolator(freqs)
+
+    # XXX see note in Tsky class for same method
+    @staticmethod
+    def _names():
+        return ("omega_p", "bandpass", )
+
+    @cached_property
+    def _interpolator(self):
+        if self._interp_type=='poly1d':
+            return np.poly1d(self._data)
+        else:
+            # if not using poly1d, then need to get some parameters for
+            # making the interp1d object
+            obj = self._data[self._obj]
+            freqs = self._data['freqs']
+
+            # use a cubic spline by default, but override this if the user
+            # specifies a different kind of interpolator
+            kind = self._interp_kwargs.pop('kind', 'cubic')
+            return interp1d(freqs, obj, kind=kind, **self._interp_kwargs)
+
+    def _check_format(self):
+        if self._obj is None:
+            raise ValueError("Please specify what type of object the " \
+                             "interpolator represents by using the `object` " \
+                             "kwarg. The currently supported object types are " \
+                             "{}.".format(INTERP_OBJECTS['1d'])
+                             )
+        else:
+            assert self._obj in INTERP_OBJECTS['1d'], \
+                    "The specified object type is not supported. The currently " \
+                    "supported object types are: {}".format(INTERP_OBJECTS['1d'])
+
+        assert self._interp_type in ('poly1d', 'interp1d'), \
+                "Interpolator choice must either be 'poly1d' or 'interp1d'."
+
+        if self._interp_type=='interp1d':
+            assert self.datafile[-4:]=='.npz', \
+                    "In order to use an 'interp1d' object, the reference file " \
+                    "must be a '.npz' file."
+            assert self._obj in self._data.keys() and 'freqs' in self._data.keys(), \
+                    "You've chosen to use an interp1d object for modeling the " \
+                    "{}. Please ensure that the `.npz` archive has the following " \
+                    "keys: 'freqs', '{}'".format(self._obj, self._obj)
```

Comment:
is the last key here supposed to just be "object" rather than `self._obj`?

Reply:
Ah, no, the way I was envisioning it was that this would add a keyword 'object' whose value specified whether the interpolation object represents a beam or a bandpass. Although I now realize this is not a good approach.